test/integration/auth/node_test.go

```diff
@@ -492,19 +501,21 @@ func TestNodeAuthorizer(t *testing.T) {
 
 	expectForbidden(t, createNode2NormalPod(nodeanonClient))
 	// mirror pod and self node lifecycle is allowed
+	expectAllowed(t, createNode2(node2Client))
+	expectAllowed(t, updateNode2Status(node2Client))
+	expectForbidden(t, createNode2MirrorPod(nodeanonClient))
+	expectForbidden(t, deleteNode2MirrorPod(nodeanonClient))
+	expectForbidden(t, createNode2MirrorPod(node1Client))
+	expectNotFound(t, deleteNode2MirrorPod(node1Client))
+	// create a pod as an admin to add object references
+	expectAllowed(t, createNode2NormalPod(superuserClient))
+
 	expectAllowed(t, createNode2MirrorPod(node2Client))
 	expectAllowed(t, deleteNode2MirrorPod(node2Client))
 	expectAllowed(t, createNode2MirrorPod(node2Client))
 	expectAllowed(t, createNode2MirrorPodEviction(node2Client))
-	expectAllowed(t, createNode2(node2Client))
-	expectAllowed(t, updateNode2Status(node2Client))
```

Comment:
why did you move a bunch of the expectations around?

Reply:
The existing order is flawed.
e.g. Node2 needs to be created ahead of pod being created on node2.
Otherwise we would get:
```
    TestNodeAuthorizer: wait.go:211: unexpected response, will retry: nodes "node2" not found
    TestNodeAuthorizer: wait.go:211: unexpected response, will retry: nodes "node2" not found
    TestNodeAuthorizer: wait.go:211: unexpected response, will retry: nodes "node2" not found
```